### ISSUE TITLE
Key columns should always appear as the left-most n columns

### ISSUE THREAD

**Issue by gmbecker** [insightsengineering/rlistings#36]
```
lsting <- as_listing(
  df = ex_adae,
  cols = c("USUBJID", "AGE", "SEX", "RACE", "ARM", "AEDECOD", "AESEV", "AETOXGR"),
  key_cols = c("ARM", "USUBJID", "AETOXGR")
)
```

```
> head(lsting, 15)
Unique Subject Identifier   Age   Sex             Race              Description of Planned Arm   Dictionary-Derived Term   Severity/Intensity   Analysis Toxicity Grade
———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————
AB12345-BRA-1-id-134        47    M               WHITE                     A: Drug X                 dcd A.1.1.1.2             MODERATE                   2           
                            47    M               WHITE                                               dcd A.1.1.1.2             MODERATE                               
                            47    M               WHITE                                               dcd B.2.1.2.1             MODERATE                   3           
                            47    M               WHITE                                               dcd D.1.1.4.2             MODERATE                               
AB12345-BRA-1-id-42         36    M     BLACK OR AFRICAN AMERICAN           A: Drug X                 dcd B.2.2.3.1               MILD                     1           
                            36    M     BLACK OR AFRICAN AMERICAN                                     dcd C.2.1.2.1             MODERATE                   2           
                            36    M     BLACK OR AFRICAN AMERICAN                                     dcd C.2.1.2.1             MODERATE                               
                            36    M     BLACK OR AFRICAN AMERICAN                                     dcd A.1.1.1.2             MODERATE                               
                            36    M     BLACK OR AFRICAN AMERICAN                                     dcd A.1.1.1.2             MODERATE                               
                            36    M     BLACK OR AFRICAN AMERICAN                                     dcd A.1.1.1.2             MODERATE                               
                            36    M     BLACK OR AFRICAN AMERICAN                                     dcd C.1.1.1.3              SEVERE                    4           
                            36    M     BLACK OR AFRICAN AMERICAN                                     dcd D.1.1.1.1              SEVERE                    5           
                            36    M     BLACK OR AFRICAN AMERICAN                                     dcd B.1.1.1.1              SEVERE                                
                            36    M     BLACK OR AFRICAN AMERICAN                                     dcd B.1.1.1.1              SEVERE                                
AB12345-BRA-1-id-93         34    F               ASIAN                     A: Drug X                 dcd D.2.1.5.3               MILD                     1          
```

**Comment by Melkiades** [comment]
If this is done:
```r
as_listing(
    df = ex_adae,
    cols = c("ARM", "USUBJID", "AETOXGR", "AGE", "SEX", "RACE", "AEDECOD", "AESEV", "AETOXGR"),
    key_cols = c("ARM", "USUBJID", "AETOXGR")
) %>% head()
```
It works. I think the output depends principally on the cols order. I wonder if this should be a feature more than a bug. I think it gives more power to order the various columns and it just needs to be documented correctly. Maybe there are special cases where ordering by `key_cols` first is not what is desired.

**Comment by Melkiades** [comment]
Note that even if values in `cols` are repeated only the first occurrence is used (which I guess is the wanted behavior)

**Comment by Melkiades** [comment]
I think this is connected with discussion #44. In my opinion, the following should be the standard possibilities:
1. Order of `key_cols` should follow a growing number of factor levels 
2. If only `key_cols` are inserted -> all cols are displayed
3. The displayed columns are always merged between `cols` and `key_cols` so to avoid mistakes
4. As a plus, I would add an argument that allows `manual_order = TRUE` (standard would be `FALSE`) in which you need to specify all the cols to be displayed in `cols`, and its specific order is maintained. Here, it being "manual", I think also repetitions should be allowed, envisioning a possible multi-page presentation where one or more columns are repeated for reference (probably very rare cases if any, but I think it makes sense to keep this possibility)

**Comment by gmbecker** [comment]
> I think this is connected with discussion #44. In my opinion, the following should be the standard possibilities:
> 
> 1. Order of `key_cols` should follow a growing number of factor levels

I don't think this is always the right order. USUBJID -> AETOXGR is a counter-example, I believe.

> 2. If only `key_cols` are inserted -> all cols are displayed

This is debatable but I'm leaning this way as well.

> 3. The displayed columns are always merged between `cols` and `key_cols` so to avoid current errors and repetitions

That is what I am leaning towards as well, yes. 

> 4. As a plus, I would add an argument that allows `manual_order = TRUE` (standard would be `FALSE`) in which you need to specify all the cols to be displayed in `cols`, and its specific order is maintained. Here, it being "manual", I think also repetitions should be allowed, envisioning a possible multi-page presentation where one or more columns are repeated for reference (probably very rare cases if any, but I think it makes sense to keep this possibility)

There are a couple of things there

a) I think the standard should be that if you specify any columns, they should appear in the order yo uspecify them in (with the caveat about key_columns always being first).

b) key_columns are (or should be, we need to test this, but by design they are) repeated across horizontal pagination. I'm not convinced we should support repetition of columns beyond that. If someone REALLY needed this they could always duplicate the column in the incoming df (rlistings would then consider them different columns that happen to have the same column label, which it wouldn't care about)

**Comment by gmbecker** [comment]
fixed in #69
